Remove usage of deprecated OSAtomic methods

LumioseSil opened the pull request:
#### 7c820f37fe12b07ef482f5057898cad2ef5c650f
<pre>
Remove usage of deprecated OSAtomic methods
<a href="https://bugs.webkit.org/show_bug.cgi?id=251865">https://bugs.webkit.org/show_bug.cgi?id=251865</a>

We should be using WTF/Atomics instead.

Reviewed by NOBODY (OOPS!).

*Source/WebCore/platform/ios/wak/WebCoreThread.mm: Removed libkern/OSAtomics import
*Source/WebKitLegacy/mac/WebView/WebView.mm: Made didDrawTiles a proper BOOL and replaced OSAtomicCompareAndSwap32 with atomicCompareExchangeStrong from WTF
*Source/WebKitLegacy/mac/WebView/WebViewData.h: Made didDrawTiles a proper BOOL
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/246bda184b8fff931db5db18cfb3d2e793deb825

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/4898 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/5023 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/5195 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/6414 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/5012 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/4891 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/5197 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/4995 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/5255 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/4983 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/5061 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/4385 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/6431 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/2549 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/4381 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/9348 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/4411 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/4455 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/6055 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/4874 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/3964 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/4374 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/8439 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/4735 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

geoffreygaren commented on Source/WebKitLegacy/mac/WebView/WebView.mm:
```diff
@@ -2312,10 +2312,8 @@ - (void)_dispatchTileDidDraw:(CALayer*)tile
 
-    ALLOW_DEPRECATED_DECLARATIONS_BEGIN
-    if (!OSAtomicCompareAndSwap32(0, 1, &_private->didDrawTiles))
+    if (!WTF::atomicCompareExchangeWeakRelaxed(&_private->didDrawTiles, NO, YES))
```
These two functions are not equivalent.

AZero13 replied:
Examining the definitions of both functions, yes they are equivalent.

geoffreygaren replied:
Can you post the definitions you examined, and define the architectures on which you think they're equivalent?

AZero13 replied:
Yes. Here 
<img width="923" alt="Screen Shot 2023-02-21 at 8 53 29 AM" src="https://user-images.githubusercontent.com/83477269/220364621-4c47d04f-6958-40f7-8176-13e5c97e9bf9.png">
<img width="923" alt="Screen Shot 2023-02-21 at 8 53 12 AM" src="https://user-images.githubusercontent.com/83477269/220364625-8de8fc00-f36b-4e99-94cf-1909000aacc2.png">
<img width="1096" alt="Screen Shot 2023-02-21 at 8 51 26 AM" src="https://user-images.githubusercontent.com/83477269/220364628-af53b42d-2cdb-459c-91f3-1c4a189151ba.png">

AZero13 replied:
I think strong is overkill, but if you deem it appropriate, I can fix that.

geoffreygaren replied:
When you say "I think strong is overkill", does that mean that you now see that these functions are not equivalent?

If so, can you explain the difference between these functions, and why you believe that difference is OK?

AZero13 replied:
Yes, one has a strong atomic compare which affects memory sequence, and the one I chose is more relaxed since the original code did not write OSAtomicCompare with the barrier.

geoffreygaren replied:
No, that's not right. Please re-read the documentation on compare_exchange_weak.

AZero13 replied:
OK I understand it now. It should be strong.

geoffreygaren reviewed (changes requested):
My comment was marked as resolved, but it is not resolved.

geoffreygaren reviewed (changes requested):
My comment is still not resolved.

LumioseSil committed:
Remove usage of deprecated OSAtomic methods
https://bugs.webkit.org/show_bug.cgi?id=251865

We should be using WTF/Atomics instead.

Reviewed by NOBODY (OOPS!).

*Source/WebCore/platform/ios/wak/WebCoreThread.mm: Removed libkern/OSAtomics import
*Source/WebKitLegacy/mac/WebView/WebView.mm: Made didDrawTiles a proper BOOL and replaced OSAtomicCompareAndSwap32 with atomicCompareExchangeStrong from WTF
*Source/WebKitLegacy/mac/WebView/WebViewData.h: Made didDrawTiles a proper BOOL
